src/js/game/LevelMVC/LevelModel.js

```diff
@@ -261,6 +261,46 @@ module.exports = class LevelModel {
     return this.controller.score;
   }
 
+  shouldRide(direction) {
+    let player = this.player;
+    let frontPosition = this.getNextRailPosition(player);
+    let frontBlock = this.actionPlane.getBlockAt(frontPosition);
+    return this.isNextRailValid(frontBlock, direction);
+  }
+
+  isNextRailValid(block, direction) {
+    if (!block) {
+      return;
+    }
+    return this.inverseDirection(block.connectionA) === direction ||
+    this.inverseDirection(block.connectionB) === direction ||
+    block.connectionA === direction ||
+    block.connectionB === direction;
+  }
+
+  inverseDirection(direction) {
+    return (direction + 2) % 4;
+  }
```

Comment:
I believe there is already a FacingDirection method for this

Reply:
So it does. But ppfffffft, why switch case when you can just add and mod? 😂